types/block.go

```diff
@@ -806,18 +829,53 @@ func (sh SignedHeader) StringIndented(indent string) string {
 
 //-----------------------------------------------------------------------------
 
-// Data contains the set of transactions included in the block
+// Data contains the data that will be erasure coded and is guaranteed to be available
+// via the data availability scheme.
+//
+// TODO: update Data.Hash() to data availability root
+// and clarify this here too.
+//
+// Data consists of:
+//  - the set of transactions included in the block
+//  - intermediate state roots computed from the Tx of the previous block
+//  - Evidence included in the block
+//  - LazyLedger (namespaced) messages included in the block
 type Data struct {
 
 	// Txs that will be applied by state @ block.Height+1.
 	// NOTE: not all txs here are valid.  We're just agreeing on the order first.
 	// This means that block.AppHash does not include these txs.
 	Txs Txs `json:"txs"`
 
+	// Intermediate state roots of the Txs included in block.Height
+	// and executed by state state @ block.Height+1.
+	//
+	// TODO: replace with a dedicated type `IntermediateStateRoot`
+	// as soon as we settle on the format / sparse Merkle tree etc
+	IntermediateStateRoots []tmbytes.HexBytes
```

Comment:
If you wonder about `tmbytes.HexBytes`: just think of it as `[]byte` (it's just for encoding really).